packages/loader/src/resource-deserialize/resources/parser/ParserContext.ts

```diff
@@ -29,10 +26,31 @@ export class ParserContext<T extends IHierarchyFile, I extends EngineObject> {
     this.resourceManager = engine.resourceManager;
   }
 
-  /**
-   * Destroy the context.
-   * @memberof ParserContext
-   */
+  addComponent(id: string, component: Component) {
+    this.components.set(id, component);
+    const waitingList = this.componentWaitingMap.get(id);
+    if (waitingList?.length) {
+      waitingList.forEach((resolve) => resolve(component));
+      this.componentWaitingMap.delete(id);
+    }
+  }
+
+  getComponentByRef(ref: IComponentRef): Promise<Component> {
```

Comment:
If a component is missing, the Promise will never resolve.

Reply:
done